### ISSUE TITLE
Can no longer cast Pandas Timestamps to timestamptz in 0.9

### ISSUE THREAD

**Issue by dylanscott** [duckdb/duckdb#9532]
### What happens?

Prior to the 0.9 release you could cast Pandas Timestamp columns to `timestamptz` resulting in the same date+time but in the configured session time zone. In version 0.9 the same cast produces a conversion error. I could not find any documentation of this in the changelog.

### To Reproduce

Minimal reproduction:

```python
import duckdb
import pandas as pd

ts_df = pd.DataFrame({ "ts": [pd.Timestamp.now()] })
query = "select cast(ts as timestamptz) as tstz from ts_df"

with duckdb.connect() as conn:
    print(conn.sql(query).df())
    conn.sql("set TimeZone = 'America/Los_Angeles'")
    print(conn.sql(query).df())
```

On version 0.8.1 this produces the output

```
                              tstz
0 2023-10-31 22:19:37.432103+00:00
                              tstz
0 2023-10-31 22:19:37.432103-07:00
```

On version 0.9.1 (and on the latest master) it results in

```
ConversionException: Conversion Error: Unimplemented type for cast (TIMESTAMP_NS -> TIMESTAMP WITH TIME ZONE)
```

### OS:

macOS, Linux

### DuckDB Version:

0.9.1, 0.9.2.dev231

### DuckDB Client:

Python

### Full Name:

Dylan Scott

### Affiliation:

Hex Technologies

### Have you tried this on the latest `main` branch?

I have tested with a main build

### Have you tried the steps to reproduce? Do they include all relevant data and configuration? Does the issue you report still appear there?

- [X] Yes, I have

**Comment by Tishj** [comment]
Hi, that's not really supposed to happen, but I understand why.

<https://github.com/duckdb/duckdb/pull/8620> has added support for non-ns timestamps, so we don't convert these to US now.
timestamp_ns -> timestamp_tz is not supported, so you'll have to add a cast from timestamp_ns to timestamp_us first

**Comment by dylanscott** [comment]
Thanks for the quick reply @Tishj. Is there any way to trigger that conversion implicitly? We have many, many existing queries written by our users that expect to be able to perform this cast, so this represents a significant breaking change for us

**Comment by Tishj** [comment]
That's fair, we could add the `TIMESTAMP_NS` -> `TIMESTAMP_TZ` cast so this will still work

**Comment by alexander-beedie** [comment]
FYI: it seems a number of similar casts now fail; I'll make a separate issue detailing them (with test cases), as I just discovered that we're blocked on upgrading to the 0.9 series by multiple failed temporal casts that previously worked in 0.8, such as:
```
Conversion Error: Unimplemented type for cast (TIMESTAMP_NS -> DATE)
Conversion Error: Unimplemented type for cast (TIMESTAMP_NS -> TIME)
Binder Error: epoch can only take temporal arguments
```

**Comment by dylanscott** [comment]
Just curious if there's been any progress here. It looks like 0.10 is still affected and is a blocker for us upgrading

**Comment by Tishj** [comment]
This PR is on hold but should fix these issues
<https://github.com/duckdb/duckdb/pull/9539>

**Comment by suchanlee** [comment]
Any chance it can go in soon and be released as part of the 0.10.0?
